### PR TITLE
fixes bug 1164936 - Removed events should not appear in the upcoming …

### PR DESCRIPTION
**Pull request by blossomica**
…assignments list.

@peterbe - I worked on this bug too.

### CODE REVIEW

File: airmozilla/manage/templates/manage/event_assignments.html [hunk 1]
```diff
@@ -54,6 +54,7 @@
     <tbody>
     {% for event in events %}
       <tr>
+       {% if not event.is_removed() %}
```
Comment (peterbe) [inline airmozilla/manage/templates/manage/event_assignments.html]: This is not the right place to do the filtering. You should do that deeper down. Like here maybe: https://github.com/blossomica/airmozilla/blob/bug1164936-Removed-events-should-not-appear-in-the-upcoming-assignments-list/airmozilla/manage/views/events.py#L1348

**Comment by blossomica** [comment]
@peterbe : I did some edits. Hopefully the last PR is the right one.

**Comment by peterbe** [comment]
No, all you need to do is change this line:
```python
qs = Event.objects.filter(start_time__gte=now)
```
to this:
```python
qs = Event.objects.scheduled().filter(start_time__gte=now)
```

**Comment by peterbe** [comment]
I know it might feel frustrating now but once you see it you'll see how obvious it is. You just need to take a tiny step back and get the idea. It's hard when so much is going on.

**Comment by blossomica** [comment]
@peterbe - Okay. I fixed that. The  other ones worked properly too so sometimes it is hard to know which method to use.

File: airmozilla/manage/views/events.py [hunk 2]
```diff
@@ -1345,7 +1345,7 @@ def __init__(self, user, count):
 
     events = []
     now = timezone.now()
-    qs = Event.objects.filter(start_time__gte=now)
+    qs = Event.objects.scheduled().filter(start_time__gte=now)
```
Comment (peterbe) [inline airmozilla/manage/views/events.py]: No, we actually just want to exclude removed ones. I would use:
```python
qs = (
    Event.objects
    .exclude(status=Event.STATUS_REMOVED)
    .filter(start_time__gte=now)
)
```

**Comment by blossomica** [comment]
@peterbe okay I fixed this. I did try something similar but I placed .exclude after .filter and it was giving me errors so that's probably a syntax mistake I did and it makes sense in some way on why I got the error that I did. Thanks for the input :)

**Comment by blossomica** [comment]
@peterbe - After second thoughts. This method doesn't work either sadly.

**Comment by peterbe** [comment]
The tests fail because of something that isn't your fault. See https://bugzilla.mozilla.org/show_bug.cgi?id=1167211

I'm going to investigate why that might be.

**Comment by peterbe** [comment]
So, another thing you need to do is prove this change with a test. Look at these lines:
https://github.com/blossomica/airmozilla/blob/bug1164936-Removed-events-should-not-appear-in-the-upcoming-assignments-list/airmozilla/manage/tests/views/test_eventassignment.py#L77-L80

It creates another event and on line 88 it asserts that it's in the `response.content`.

Change that event to `event.status = Event.STATUS_REMOVED` then do what's done on line 86 again and this time assert that 'Other title' is NOT in the `response.content`.

You'll need to change line 77 to something like:
```
other_event = Event.objects.create(....
```
So you can change its status later.

**Comment by blossomica** [comment]
@peterbe - Finally I think this resolved? Let me know if it is time for squashing.

**Comment by peterbe** [comment]
I'll do some manual testing too.

**Comment by blossomica** [comment]
All right, let me know how it goes. This PR is okay in fact. There is another one (the one with processing/scheduled) that gives me the tweets error.  https://github.com/mozilla/airmozilla/pull/310

**Comment by peterbe** [comment]
r+

**Comment by blossomica** [comment]
@peterbe  - Done

**Pull request merged**